packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -219,38 +239,73 @@ export default class GoogleCalendarService implements Calendar {
       "primary";
 
     try {
-      const event = await calendar.events.insert({
-        calendarId: selectedCalendar,
-        requestBody: payload,
-        conferenceDataVersion: 1,
-        sendUpdates: "none",
-      });
+      let event;
+      let recurringEventId = null;
+      if (calEventRaw.existingRecurringEvent) {
+        recurringEventId = calEventRaw.existingRecurringEvent.recurringEventId;
+        const recurringEventInstances = await calendar.events.instances({
+          calendarId: selectedCalendar,
+          eventId: calEventRaw.existingRecurringEvent.recurringEventId,
+        });
+        event = recurringEventInstances.data.items![calEventRaw.existingRecurringEvent.eventIndex];
```

Comment:
This seems like a very risky approach to me. We are using the index of the occurence as per the number of occurences we have in our system. That index is used to get the event in Google Calendar. Google Calendar and Cal.com can easily be out of sync and we can endup accidentally referring to wrong event in Google Calendar.

Reply:
Changed so that a match is done by comparing start times